### DISCUSSION TITLE
WEBRTC release for macOS

### DISCUSSION THREAD

**Discussion by dhandby** [SafeExamBrowser/seb-mac#115] [Q&A]
Hi Guys,

We are just curious as to when the release of webrtc for mac might be coming along or even the next release of SEB for MAC? I know screensharing is going to be a problem due to apple's privacy settings etc and trying to find a workaround.

Thanks,
Dean

**Comment by danschlet** [comment]
Hi Dean,

I just released a [beta version of SEB 3.0 for macOS](https://github.com/SafeExamBrowser/seb-mac/releases/tag/3.0b1) with WebRTC support. There are a few minor features missing and it needs proper testing, but I expect the final release soon.

Yes, unfortunately Apple only supports getUserMedia in the embeddable WKWebView browser view, not getDisplayMedia. I will investigate in the coming weeks if I could somehow add support for that on my own (it works in Safari, maybe it is in the public WebKit source code and I could just build my own copy of the WebKit framework, which I did in the past). Or maybe I could implement some kind of a Javascript bridge and use the open source WebRTC framework (library).

Daniel

**Reply by dhandby** [reply]
Thanks Daniel.

We receive the below message using the new version (beta release). Using macOS 11.4 and 11.5.2

Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'navigator.mediaDevices.getUserMedia')

Thanks for working on this for us though!

**Reply by danschlet** [reply]
What are you using to test this? If you can share a test link, I can have a look.

Did you read the prerequisites for SEB using the modern WebView? If you're not on macOS Big Sur 11.1 or later and if your settings are not correct, SEB by default uses the classic WebView for the main browser window where the exam system is running in. 

Moodle for example will require an update of the SEB integration so you could use a plugin requiring WebRTC. Currently you could only use a service requiring WebRTC with Moodle if it's loaded in a separate browser window and has a different host address in the URL than Moodle. Or you need to switch off the ConfigKey/BrowserExamKey check in Moodle.

**Reply by dhandby** [reply]
Hi Daniel,

I managed to adjust the settings as you said in the macOS configuration tool by turning off ConfigKey/BrowserExamKey check.

SEB prompts for microphone and webcam now.

Although during my test I was not using moodle at all and going straight into our proctoring tool website.

Thanks,
Dean
